hedged.go

```diff
@@ -88,21 +161,31 @@ type hedgedTransport struct {
 	rt      http.RoundTripper
 	timeout time.Duration
 	upto    int
+	next    NextFn
 	metrics *Stats
 }
 
 func (ht *hedgedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	mainCtx := req.Context()
 
-	timeout := ht.timeout
+	upto, timeout := ht.upto, ht.timeout
+	if ht.next != nil {
+		upto, timeout = ht.next()
+	}
+
+	// no hedged requests, just a regular one.
+	if upto == 0 {
```

Comment:
Technically possible for a buggy implementation to return <0 and that will lead to a panic later on in this function because I think you cannot do `make(chan int, -1)`, for example :smile:

Reply:
Ahaha, yeah, it should be `upto <= 0`. Gonna fix that and release `v0.9.1`. Thank you.

Reply:
Done https://github.com/cristalhq/hedgedhttp/pull/53